spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dao/XdJdbcSearchableJobExecutionDao.java

```diff
@@ -0,0 +1,160 @@
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.springframework.batch.admin.service.JdbcSearchableJobExecutionDao;
+import org.springframework.batch.admin.service.SearchableJobExecutionDao;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.repository.dao.JdbcJobExecutionDao;
+import org.springframework.batch.item.database.Order;
+import org.springframework.batch.item.database.PagingQueryProvider;
+import org.springframework.batch.item.database.support.SqlPagingQueryProviderFactoryBean;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.incrementer.AbstractDataFieldMaxValueIncrementer;
+import org.springframework.util.Assert;
+
+/**
+ * @author Dave Syer
+ * @author Michael Minella
+ * @author Gunnar Hillert
+ *
+ */
+public class XdJdbcSearchableJobExecutionDao extends JdbcSearchableJobExecutionDao {
+
+	private static final String FIELDS = "E.JOB_EXECUTION_ID, E.START_TIME, E.END_TIME, E.STATUS, E.EXIT_CODE, E.EXIT_MESSAGE, "
+			+ "E.CREATE_TIME, E.LAST_UPDATED, E.VERSION, I.JOB_INSTANCE_ID, I.JOB_NAME";
+
+	private PagingQueryProvider allExecutionsPagingQueryProvider;
+
+	private PagingQueryProvider childJobExecutionsPagingQueryProvider;
+
+	private DataSource dataSource;
+
+	/**
+	 * @param dataSource the dataSource to set
+	 */
+	@Override
+	public void setDataSource(DataSource dataSource) {
+		this.dataSource = dataSource;
+		super.setDataSource(dataSource);
+	}
+
+	/**
+	 * @see JdbcJobExecutionDao#afterPropertiesSet()
+	 */
+	@Override
+	public void afterPropertiesSet() throws Exception {
+
+		Assert.state(dataSource != null, "DataSource must be provided");
+
+		if (getJdbcTemplate() == null) {
+			setJdbcTemplate(new JdbcTemplate(dataSource));
+		}
+		setJobExecutionIncrementer(new AbstractDataFieldMaxValueIncrementer() {
+
+			@Override
+			protected long getNextKey() {
+				return 0;
+			}
+		});
+
+		final String subQuery = getQuery(
+				"not exists (select * from %PREFIX%JOB_EXECUTION_PARAMS a where key_name = 'xd_parent_execution_id' and E.JOB_EXECUTION_ID=a.JOB_EXECUTION_ID)");
+		allExecutionsPagingQueryProvider = getPagingQueryProvider(null, subQuery);
+
+		final String childJobExecutionsSubQuery = getQuery(
+				"exists (select * from %PREFIX%JOB_EXECUTION_PARAMS a where key_name = 'xd_parent_execution_id' and E.JOB_EXECUTION_ID=a.JOB_EXECUTION_ID and long_val=?)");
+		childJobExecutionsPagingQueryProvider = getPagingQueryProvider(null, childJobExecutionsSubQuery);
+
+		super.afterPropertiesSet();
+
+	}
+
+	/**
+	 * @return a {@link PagingQueryProvider} with a where clause to narrow the
+	 * query
+	 * @throws Exception
+	 */
+	private PagingQueryProvider getPagingQueryProvider(String fromClause, String whereClause) throws Exception {
```

Comment:
If the `fromClause` is always null and set inside the method, why can't we just remove this method as argument?